docs/sources/developers/plugins/build-a-logs-data-source-plugin.md

```diff
@@ -51,7 +58,9 @@ That's all you need to start returning log data from your data source. Go ahead
 
 Congratulations, you just wrote your first logs data source plugin! Next, let's look at a couple of features that can further improve the experience for the user.
 
-### (Optional) Add preferred visualisation type hint to the data frame
+## Enhance your logs data source plugin with optional features
+
+### Add a preferred visualisation type hint to the data frame
```

Comment:
If this is the only optional feature, then maybe add Optional: to this heading and remove the previous Enhance your logs data source plugins... heading entirely.